### PR TITLE
Maven Daemon CLIng

### PR DESCRIPTION
**Pull request by cstamas**
Dropped the MavenCLI mutilated copy, using CLIng.

This PR also removes test repeat and use of mrm plugin, just to simplify the build. Also fixes classpath by excluding maven logging provider (that is excluded from final distro as well) as mvnd provides its own provider.

### CODE REVIEW

**Comment by cstamas** [comment]
Pretty much same results as for me locally:
```
Error:  Tests run: 41, Failures: 9, Errors: 1, Skipped: 3
```

**Comment by cstamas** [comment]
Locally
```
[ERROR] Tests run: 41, Failures: 8, Errors: 0, Skipped: 3
```

**Comment by cstamas** [comment]
With https://github.com/apache/maven/pull/1776
```
[ERROR] Tests run: 41, Failures: 7, Errors: 0, Skipped: 3
```

**Comment by cstamas** [comment]
Slowly but mvnd is giving up :smile: 

```
[ERROR] Tests run: 41, Failures: 4, Errors: 1, Skipped: 3
```

**Comment by cstamas** [comment]
Score
```
[ERROR] Tests run: 41, Failures: 4, Errors: 0, Skipped: 3
```

**Comment by cstamas** [comment]
At least two ITs are NPE-ing due unusual IT project layout @gnodet is tinkering about those. Personally, I'd not change ITs, Maven is too rigid IMHO.

**Comment by cstamas** [comment]
CI confirms my local results, failed ITs with reason:
* EnvironmentTest.cleanInstall -- path cannot be null https://issues.apache.org/jira/browse/MNG-8288
* MavenConfTest -- An API incompatibility was encountered while executing org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate: java.lang.ExceptionInInitializerError: null
* NewManagedModuleTest -- this one is weird Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.mvndaemon.mvnd.test.new-managed-module:new-managed-module-bom:pom:0.0.1-SNAPSHOT in mrm-maven-plugin (http://localhost:37133/)
* UpgradesInBomTest -- path cannot be null https://issues.apache.org/jira/browse/MNG-8288

**Comment by cstamas** [comment]
Removed use of mrm plugin, test repeats, and fixed classpath by excluding maven logging provider (that is excluded from final distro as well).

**Comment by cstamas** [comment]
@gnodet @ppalaga it's yours, please take a look

File: daemon/src/main/java/org/apache/maven/cli/DaemonMavenParser.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                (DaemonMavenOptions) context.options);
+    }
+
+    @Override
+    protected Path getRootDirectory(LocalContext context) throws ParserException {
+        Map<String, String> env = context.parserRequest
+                .lookup()
+                .lookup(DaemonMavenCling.Environment.class)
+                .getEnvironment();
+        System.setProperty("maven.multiModuleProjectDirectory", env.get("maven.multiModuleProjectDirectory"));
```
Comment (cstamas) [inline daemon/src/main/java/org/apache/maven/cli/DaemonMavenParser.java:61]: This should be moved or reworked: parser should not alter state (we undid same thing in maven-cli)

File: daemon/src/main/java/org/apache/maven/cli/DaemonMavenParser.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    protected List<CoreExtension> readCoreExtensionsDescriptor(LocalContext context) {
+        return Stream.of(Environment.MVND_CORE_EXTENSIONS.asString().split(";"))
+                .filter(s -> s != null && !s.isEmpty())
+                .map(s -> {
+                    String[] parts = s.split(":");
+                    return CoreExtension.newBuilder()
+                            .groupId(parts[0])
+                            .artifactId(parts[1])
+                            .version(parts[2])
+                            .build();
```
Comment (cstamas) [inline daemon/src/main/java/org/apache/maven/cli/DaemonMavenParser.java:101]: This needs to be extended, as extensions.xml may have more options, like classloading?

Comment (cstamas) [reply daemon/src/main/java/org/apache/maven/cli/DaemonMavenParser.java:101]: This is for another PR, as this PR did not alter anything in this area

**Comment by cstamas** [comment]
Mac intel had "java.net.ConnectException: HTTP connect timed out"

**Review by gnodet — APPROVED** [review]

**Pull request merged**